clusterversion: remove BinaryVersionKey/BinaryMinSupportedVersionKey

RaduBerinde opened the pull request:
Remove `BinaryVersionKey` and `BinaryMinSupportedVersionKey` in favor of `Latest` and `MinSupported`.

Informs: #112629
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114172)
<!-- Reviewable:end -->

RaduBerinde committed:
clusterversion: remove BinaryVersionKey/BinaryMinSupportedVersionKey

Remove `BinaryVersionKey` and `BinaryMinSupportedVersionKey` in favor
of `Latest` and `MinSupported`.

Informs: #112629
Release note: None

RaduBerinde commented:
bors r+
